src/utils/mapsheet.ts

```diff
@@ -19,89 +47,128 @@ export interface Size {
-  if (out.gridSize === 500) {
-    out.y = Number(match[3].slice(0, 3));
-    out.x = Number(match[3].slice(3));
-  } else {
-    out.y = Number(match[3].slice(0, 2));
-    out.x = Number(match[3].slice(2));
-  }
-  if (isNaN(out.gridSize) || isNaN(out.x) || isNaN(out.y)) return null;
-
-  const origin = MapSheet.offset(out.mapSheet);
-  if (origin == null) return null;
-
-  const tileOffset = MapSheet.tileOffset(out.gridSize, out.x, out.y);
-  out.expected.x = origin.x + tileOffset.x;
-  out.expected.y = origin.y - tileOffset.y;
-  out.expected.width = tileOffset.width;
-  out.expected.height = tileOffset.height;
-  return out;
-}
-
 const charA = 'A'.charCodeAt(0);
 const charS = 'S'.charCodeAt(0);
-function* iterateMapSheets(): Generator<string> {
-  for (let first = 0; first < 3; first++) {
-    for (let second = 0; second < 26; second++) {
-      if (first === 0 && second < charS - charA) continue;
-      const mapSheet = `${String.fromCharCode(charA + first)}${String.fromCharCode(charA + second)}`;
-      if (Skipped.has(mapSheet)) continue;
-
-      yield mapSheet;
-      if (mapSheet === MapSheet.code.end) return;
-    }
-  }
-}
-
-function mapSheetOffset(sheetCode: string): { x: number; y: number } | null {
-  const ms = sheetCode.slice(0, 2);
-  const x = Number(sheetCode.slice(2));
-
-  const baseYOffset = charS - charA;
-  const firstLetterOffset = (ms.charCodeAt(0) - charA) * 26;
-  const secondLetterOffset = ms.charCodeAt(1) - charA;
-
-  let y = firstLetterOffset + secondLetterOffset - baseYOffset;
-
-  // There are three missing map sheets
-  if (ms > 'CI') y -= 3;
-  else if (ms > 'BO') y -= 2;
-  else if (ms > 'BI') y--;
-
-  return { x: MapSheet.width * x + MapSheet.origin.x, y: MapSheet.origin.y - MapSheet.height * y };
-}
-
-export function tileOffset(gridSize: number, x: number, y: number): Bounds {
-  const scale = gridSize / MapSheet.scale;
-  const offsetX = MapSheet.width * scale;
-  const offsetY = MapSheet.height * scale;
-  return { x: (x - 1) * offsetX, y: (y - 1) * offsetY, width: offsetX, height: offsetY };
-}
 
 const Skipped = new Set(['BI', 'BO', 'CI']);
 
+/***
+ * Topographic 1:50k map sheet calculator
+ *
+ * Useful for working with LINZ's Topo 1:50k mapsheet names and any of the tile indexes
+ *
+ * - https://data.linz.govt.nz/layer/104687-nz-150k-tile-index/ 1:50,000
+ * - https://data.linz.govt.nz/layer/104690-nz-110k-tile-index/ 1:10,000
+ * - https://data.linz.govt.nz/layer/104691-nz-15k-tile-index/  1:5,000
+ * - https://data.linz.govt.nz/layer/106966-nz-12k-tile-index/  1:2,000
+ * - https://data.linz.govt.nz/layer/104692-nz-11k-tile-index/  1:1,000
+ * - https://data.linz.govt.nz/layer/106965-nz-1500-tile-index/ 1:500
+ **/
 export const MapSheet = {
+  /** Width of Topo 1:50k mapsheets in meters */
   height: 36_000,
+  /** Height of Topo 1:50k mapsheets in meters */
   width: 24_000,
+  /** Base scale Topo 1:50k mapsheets in meters*/
   scale: 50_000,
+  /** Map Sheets start at AS and end at CK */
```

Comment:
```suggestion
  /** Map Sheets start at AS (https://www.linz.govt.nz/sites/default/files/north-island.jpg) and end at CK (https://www.linz.govt.nz/sites/default/files/south-island.jpg) */
```

Reply:
Is there a  page these came from, I hesitant to linked directly to random images

Reply:
https://www.linz.govt.nz/products-services/maps/new-zealand-topographic-maps/topo50-map-chooser/topo50-sheet-index

Reply:
this is a good link! we should link to it somewhere in this file.